src/api/Nhs.Appointments.Persistance/BookingCosmosDocumentStore.cs

```diff
@@ -16,11 +17,30 @@ public BookingCosmosDocumentStore(ITypedDocumentCosmosStore<BookingDocument> boo
         _indexStore = indexStore;
     }
            
-    public Task<IEnumerable<Booking>> GetInDateRangeAsync(string site, DateTime from, DateTime to)
+    public Task<IEnumerable<Booking>> GetInDateRangeAsync(DateTime from, DateTime to, string site)
     {
         return _bookingStore.RunQueryAsync<Booking>(b => b.Site == site && b.From >= from && b.From <= to);
-    }       
-    
+    }
+
+    public async Task<IEnumerable<Booking>> GetCrossSiteAsync(DateTime from, DateTime to)
+    {
+        var bookingIndexDocuments = await _indexStore.RunQueryAsync<BookingIndexDocument>(i => i.From >= from && i.From <= to);
+        var grouped = bookingIndexDocuments.GroupBy(i => i.Site);
+
+        var concurrentResults = new ConcurrentBag<IEnumerable<Booking>>();
+
+        await Parallel.ForEachAsync(grouped, async (group, _) =>
+        {
+            var bookings = await GetInDateRangeAsync(group.Min(g => g.From), group.Max(g => g.From), group.Key);
+            concurrentResults.Add(bookings);
+        });
+
+        var result = new List<Booking>();
```

Comment:
The last 3 lines could be replaced with concurrentResults.SelectMany(x => x) (with .ToList()) if you absolutely wanted to copy the results into a new list